### PR TITLE
patch name database

### PR DESCRIPTION
**Pull request by jpcima**
#63

Add the database of patch names for all MIDI standards.

Not connected yet with the bank editor, for now I only implemented substitution methods of old lookups, and marked with deprecation attribute.

### CODE REVIEW

**Comment by Wohlstand** [comment]
I have took this and I have tried to implement the usage: https://github.com/Wohlstand/OPN2BankEditor/tree/wip-program-names
Unfortunately, the work didn't finished as I am tired now... You can continue the work yourself if you want, as tomorrow I'll go to my job and I will come back to my home in evenining...

**Comment by jpcima** [comment]
It will now support new names to an extent.
The menu choice will be considered. For now, I have considered GS and Sound Canvas as a single set.

There remains to do some things:
- make use of the bank name information, that the data set also has
- to update instrument names on change of bank selection, or change of either LSB/MSB
there is a problem currently on reloading from changed `bank_no`, because it will go into a loop and crash. (`bank_no` wants to `reloadInstruments`, `reloadInstruments` wants to change `bank_no`, etc..)

File: src/bank_editor.ui [hunk 1]
```diff
@@ -3810,15 +3810,15 @@ Available from 0 to 127</string>
...
+    <string>Roland GS</string>
    </property>
   </action>
   <action name="actionStandardXG">
    <property name="checkable">
     <bool>true</bool>
    </property>
    <property name="text">
-    <string notr="true">XG</string>
+    <string>Yamaha XG</string>
```
Comment (Wohlstand) [inline src/bank_editor.ui]: Reason why I added `notr="true"` those vaules are really non-translatable proper names.

**Comment by Wohlstand** [comment]
Checked this around, looks like fine! :wink: :fox_face:

**Pull request merged**